<a href="https://colab.research.google.com/github/argg1006/TRABAJOS/blob/main/EXAMENISI.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [34]:
import pandas as pd
df= pd.read_csv('WA_Fn-UseC_-HR-Employee-Attrition.csv')

#KPI 1

total_empleados=df['EmployeeNumber'].count()
empleadosDespedidos=df[df['Attrition']=='Yes']['Attrition'].count()
RotacionPersonal =(empleadosDespedidos/total_empleados)*100

print("ROTACIÓN LABORAL")
print(f"Total de empleados:{total_empleados}")
print(f"Empleados que ya no trabajan:{empleadosDespedidos}")
print(f"La rotación laboral es:{RotacionPersonal:.2f}%")




ROTACIÓN LABORAL
Total de empleados:1470
Empleados que ya no trabajan:237
La rotación laboral es:16.12%


Como resultado de la rotación laboral, se haya un 16.12%, en un total de 1470 empleados, tomando en cuenta que una tasa de aceptación laboral va entre el 10% y 15%, nos muestra que hay que preocuparnos por este porcentaje que no es favorable.

In [35]:
#KPI 2

IngresoPromedio=df.groupby('Department')['MonthlyRate'].mean().round(2)
display(IngresoPromedio)

,MonthlyRate
Department,
Human Resources,13492.98
Research & Development,14284.87
Sales,14489.79


El análisis respectivo a este resultado nos da que el menor ingreso dado para un departamento es al de Recursos Humanos, a su vez, el que cuenta con mayor ingreso es el de Sales.

In [38]:
#KPI 3

IndiceBalanceVida=df.groupby('JobRole')['WorkLifeBalance'].mean().round(2)
display(IndiceBalanceVida)

,WorkLifeBalance
JobRole,
Healthcare Representative,2.70
Human Resources,2.92
Laboratory Technician,2.72
Manager,2.77
Manufacturing Director,2.77
Research Director,2.86
Research Scientist,2.68
Sales Executive,2.80
Sales Representative,2.89


Los cargos que cuentan con el menor indice es el de Heathcare Representative, mientras que Human Resources cuenta con el mayor indice, mientras mas bajo sea el indice más desgante este cuenta.

In [40]:

#KPI 4
trabajando=df[df['Attrition']=='No'].copy()
trabajando['BrechaE']=trabajando['YearsInCurrentRole']-trabajando['YearsSinceLastPromotion']
Promedio=trabajando['BrechaE'].mean()
print(f"Brecha Promedio: {Promedio:.2f}")

Brecha Promedio: 2.25


Los empleados actuales llevan un promedio de 2.25 años sin ser ascendidos. Esto puede provocar un desgaste en la producción e igual una desmotivación.